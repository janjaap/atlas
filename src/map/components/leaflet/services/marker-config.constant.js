import {
  geoSearchType,
  dataSelectionType,
  detailPointType,
  straatbeeldPersonType,
  straatbeeldOrientationType
} from './icons.constant';

const markerConfig = {
  [geoSearchType]: { requestFocus: true },
  [dataSelectionType]: {},
  [detailPointType]: { requestFocus: true },
  [straatbeeldPersonType]: { requestFocus: true },
  [straatbeeldOrientationType]: { requestFocus: true }
};
export default markerConfig;
