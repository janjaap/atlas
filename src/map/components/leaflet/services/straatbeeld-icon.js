import { Icon } from 'leaflet';

import ICON_CONFIG from './icon-config.constant';

export const straatbeeldPersonIcon = () => new Icon({
  ...ICON_CONFIG.STRAATBEELD_PERSON,
  popupAnchor: null,
  shadowUrl: null,
  shadowSize: null,
  shadowAnchor: null,
  className: 'straatbeeld_person'
});

export const straatbeeldOrientationIcon = () => new Icon({
  ...ICON_CONFIG.STRAATBEELD_ORIENTATION,
  popupAnchor: null,
  shadowUrl: null,
  shadowSize: null,
  shadowAnchor: null,
  className: 'straatbeeld_orientation'
});
