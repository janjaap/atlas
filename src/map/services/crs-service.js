import L from 'leaflet';
import proj4 from 'proj4'; //eslint-disable-line
import proj4leaflet from 'proj4leaflet'; //eslint-disable-line

import CRS_CONFIG from './crs-config.constant';

function getCrs() {
  const rdSettings = CRS_CONFIG.RD;
  rdSettings.transformation.bounds =
    L.bounds.apply(null, CRS_CONFIG.RD.transformation.bounds);
  const crs = new L.Proj.CRS(
          rdSettings.code,
          rdSettings.projection,
          rdSettings.transformation
      );

  crs.distance = L.CRS.Earth.distance;
  crs.R = CRS_CONFIG.EARTH_RADIUS;

  return crs;
}

export const getRdObject = () => ({
  type: 'name',
  properties: {
    name: CRS_CONFIG.RD.code
  }
});

export default getCrs;
