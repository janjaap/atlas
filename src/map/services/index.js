import mapLayers from './map-layers/map-layers.config';
import mapPanelLayers from './map-panel-layers/map-panel-layers.config';
import mapBaseLayers from './map-base-layers/map-base-layers.config';

const TIMEOUT = 100;

export function getMapBaseLayers() {
  return new Promise((resolve) => {
    setTimeout(() => resolve(mapBaseLayers), TIMEOUT);
  });
}

export function getMapLayers() {
  return new Promise((resolve) => {
    setTimeout(() => resolve(mapLayers), TIMEOUT);
  });
}

export function getPanelLayers() {
  return new Promise((resolve) => {
    setTimeout(() => resolve(mapPanelLayers), TIMEOUT);
  });
}
