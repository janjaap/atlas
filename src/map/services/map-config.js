import BOUNDING_BOX from './bounding-box.constant';
import ENVIRONMENT, { ENVIRONMENTS } from '../../shared/environment';
import getCrs from './crs-service';

const BOUNDS = [
  BOUNDING_BOX.COORDINATES.southWest,
  BOUNDING_BOX.COORDINATES.northEast
];

const defaultConfig = {
  BASE_LAYER_OPTIONS: {
    tms: true,
    minZoom: 8,
    maxZoom: 16,
    bounds: BOUNDS
  },
  MAP_OPTIONS: {
    crs: getCrs(),
    maxBounds: BOUNDS,
    // 1.0 makes the bounds fully solid, preventing the user from dragging outside the bounds
    maxBoundsViscosity: 1.0,
    bounceAtZoomLimits: false,
    attributionControl: false,
    zoomControl: false
  },
  OVERLAY_OPTIONS: {
    identify: false,
    format: 'image/png',
    transparent: true
  },
  SCALE_OPTIONS: {
    position: 'bottomright',
    metric: true,
    imperial: false
  },
  ZOOM_OPTIONS: {
    position: 'bottomright',
    zoomInTitle: 'Inzoomen',
    zoomOutTitle: 'Uitzoomen'
  },
  DEFAULT_ZOOM_HIGHLIGHT: 14
};


const environmentConfig = {
  [ENVIRONMENTS.PRODUCTION]: {
    BASE_LAYER_OPTIONS: {
      subdomains: ['t1', 't2', 't3', 't4']
    },
    OVERLAY_ROOT: 'https://map.data.amsterdam.nl/'
  },
  [ENVIRONMENTS.PRE_PRODUCTION]: {
    BASE_LAYER_OPTIONS: {
      subdomains: ['t1', 't2', 't3', 't4']
    },
    OVERLAY_ROOT: 'https://map.data.amsterdam.nl/'
  },
  [ENVIRONMENTS.ACCEPTANCE]: {
    BASE_LAYER_OPTIONS: {
      subdomains: ['acc.t1', 'acc.t2', 'acc.t3', 'acc.t4']
    },
    OVERLAY_ROOT: 'https://acc.map.data.amsterdam.nl/'
  },
  [ENVIRONMENTS.DEVELOPMENT]: {
    BASE_LAYER_OPTIONS: {
      subdomains: ['acc.t1', 'acc.t2', 'acc.t3', 'acc.t4']
    },
    OVERLAY_ROOT: 'https://acc.map.data.amsterdam.nl/'
  }
};

const MAP_CONFIG = {
  ...defaultConfig,
  BASE_LAYER_OPTIONS: {
    ...defaultConfig.BASE_LAYER_OPTIONS,
    ...environmentConfig[ENVIRONMENT].BASE_LAYER_OPTIONS
  },
  OVERLAY_ROOT: environmentConfig[ENVIRONMENT].OVERLAY_ROOT
};

export default MAP_CONFIG;
