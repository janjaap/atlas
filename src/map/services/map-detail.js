import adressenLigplaats from '../../shared/services/adressen/adressen-ligplaats';
import adressenNummeraanduiding from '../../shared/services/adressen/adressen-nummeraanduiding';
import adressenOpenbareRuimte from '../../shared/services/adressen/adressen-openbare-ruimte';
import adressenPand from '../../shared/services/adressen/adressen-pand';
import adressenStandplaats from '../../shared/services/adressen/adressen-standplaats';
import adressenVerblijfsobject from '../../shared/services/adressen/adressen-verblijfsobject';
import bedrijfsinvesteringszone from '../../shared/services/bedrijfsinvesteringszone/bedrijfsinvesteringszone';
import explosievenGevrijwaardGebied from '../../shared/services/explosieven/explosieven-gevrijwaard-gebied';
import explosievenInslag from '../../shared/services/explosieven/explosieven-inslag';
import explosievenUitgevoerdOnderzoek from '../../shared/services/explosieven/explosieven-uitgevoerd-onderzoek';
import explosievenVerdachtGebied from '../../shared/services/explosieven/explosieven-verdacht-gebied';
import gebiedenBouwblok from '../../shared/services/gebieden/gebieden-bouwblok';
import gebiedenBuurt from '../../shared/services/gebieden/gebieden-buurt';
import gebiedenGebiedsgerichtWerken from '../../shared/services/gebieden/gebieden-gebiedsgericht-werken';
import gebiedenGrootstedelijk from '../../shared/services/gebieden/gebieden-grootstedelijk';
import gebiedenStadsdeel from '../../shared/services/gebieden/gebieden-stadsdeel';
import gebiedenUnesco from '../../shared/services/gebieden/gebieden-unesco';
import gebiedenWijk from '../../shared/services/gebieden/gebieden-wijk';
import grondexploitatie from '../../shared/services/grondexploitatie/grondexploitatie';
import kadastraalObject from '../../shared/services/kadastraal-object/kadastraal-object';
import meetbout from '../../shared/services/meetbout/meetbout';
import monument from '../../shared/services/monument/monument';
import napPeilmerk from '../../shared/services/nap-peilmerk/nap-peilmerk';
import vestiging from '../../shared/services/vestiging/vestiging';

export const maxDisplayValuesPerProperty = 5;

// NOTE: beware, if FETCH_DETAIL action payload includes endpointTypes string
// then magic happens in '/src/reducers/details.js'
export const endpointTypes = {
  adressenLigplaats: 'bag/ligplaats/',
  adressenNummeraanduiding: 'bag/nummeraanduiding/',
  adressenOpenbareRuimte: 'bag/openbareruimte/',
  adressenPand: 'bag/pand/',
  adressenStandplaats: 'bag/standplaats/',
  adressenVerblijfsobject: 'bag/verblijfsobject/',
  bedrijfsinvesteringszone: 'vsd/biz/',
  explosievenGevrijwaardGebied: 'milieuthemas/explosieven/gevrijwaardgebied/',
  explosievenInslag: 'milieuthemas/explosieven/inslagen/',
  explosievenUitgevoerdOnderzoek: 'milieuthemas/explosieven/uitgevoerdonderzoek/',
  explosievenVerdachtGebied: 'milieuthemas/explosieven/verdachtgebied/',
  gebiedenBouwblok: 'gebieden/bouwblok/',
  gebiedenBuurt: 'gebieden/buurt/',
  gebiedenGebiedsgerichtWerken: 'gebieden/gebiedsgerichtwerken/',
  gebiedenGrootstedelijk: 'gebieden/grootstedelijkgebied/',
  gebiedenStadsdeel: 'gebieden/stadsdeel/',
  gebiedenUnesco: 'gebieden/unesco/',
  gebiedenWijk: 'gebieden/buurtcombinatie/',
  grondexploitatie: 'grondexploitatie/project/',
  kadastraalObject: 'brk/object/',
  meetbout: 'meetbouten/meetbout/',
  monument: 'monumenten/monumenten/',
  napPeilmerk: 'nap/peilmerk/',
  vestiging: 'handelsregister/vestiging/'
};

const servicesByEndpointType = {
  [endpointTypes.adressenLigplaats]: { fetch: adressenLigplaats },
  [endpointTypes.adressenNummeraanduiding]: { fetch: adressenNummeraanduiding },
  [endpointTypes.adressenOpenbareRuimte]: { fetch: adressenOpenbareRuimte },
  [endpointTypes.adressenPand]: { fetch: adressenPand },
  [endpointTypes.adressenStandplaats]: { fetch: adressenStandplaats },
  [endpointTypes.adressenVerblijfsobject]: { fetch: adressenVerblijfsobject },
  [endpointTypes.bedrijfsinvesteringszone]: { fetch: bedrijfsinvesteringszone },
  [endpointTypes.explosievenGevrijwaardGebied]: { fetch: explosievenGevrijwaardGebied },
  [endpointTypes.explosievenInslag]: { fetch: explosievenInslag },
  [endpointTypes.explosievenUitgevoerdOnderzoek]: { fetch: explosievenUitgevoerdOnderzoek },
  [endpointTypes.explosievenVerdachtGebied]: { fetch: explosievenVerdachtGebied },
  [endpointTypes.gebiedenBouwblok]: { fetch: gebiedenBouwblok },
  [endpointTypes.gebiedenBuurt]: { fetch: gebiedenBuurt },
  [endpointTypes.gebiedenGebiedsgerichtWerken]: { fetch: gebiedenGebiedsgerichtWerken },
  [endpointTypes.gebiedenGrootstedelijk]: { fetch: gebiedenGrootstedelijk },
  [endpointTypes.gebiedenStadsdeel]: { fetch: gebiedenStadsdeel },
  [endpointTypes.gebiedenUnesco]: { fetch: gebiedenUnesco },
  [endpointTypes.gebiedenWijk]: { fetch: gebiedenWijk },
  [endpointTypes.grondexploitatie]: { fetch: grondexploitatie, authScope: 'GREX/R' },
  [endpointTypes.kadastraalObject]: { fetch: kadastraalObject },
  [endpointTypes.meetbout]: { fetch: meetbout },
  [endpointTypes.monument]: { fetch: monument },
  [endpointTypes.napPeilmerk]: { fetch: napPeilmerk },
  [endpointTypes.vestiging]: { fetch: vestiging, authScope: 'HR/R' }
};

export default function fetchDetail(endpoint, user) {
  const endpointType = Object.keys(servicesByEndpointType).find((type) => endpoint.includes(type));
  const endpointConfig = endpointType && servicesByEndpointType[endpointType];
  const fetchFn = endpointConfig && endpointConfig.fetch;
  const authScope = endpointConfig && endpointConfig.authScope;
  return fetchFn && (!authScope || user.scopes.includes(authScope)) &&
    fetchFn(endpoint);
}

window.mapPreviewPanelDetailEndpointTypes = endpointTypes;
