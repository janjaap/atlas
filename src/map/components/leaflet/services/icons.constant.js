import searchIcon from './search-icon';
import { straatbeeldPersonIcon, straatbeeldOrientationIcon } from './straatbeeld-icon';
import detailIcon from './detail-icon';
import dataSelectionIcon from './data-selection-icon';

export const geoSearchType = 'geoSearchType';
export const dataSelectionType = 'dataSelectionType';
export const detailPointType = 'detailPointType';
export const straatbeeldPersonType = 'straatbeeldPersonType';
export const straatbeeldOrientationType = 'straatbeeldOrientationType';

const icons = {
  [geoSearchType]: searchIcon,
  [dataSelectionType]: dataSelectionIcon,
  [detailPointType]: detailIcon,
  [straatbeeldPersonType]: straatbeeldPersonIcon,
  [straatbeeldOrientationType]: straatbeeldOrientationIcon
};
export default icons;
