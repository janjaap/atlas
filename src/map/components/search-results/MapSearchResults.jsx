import React from 'react';
import PropTypes from 'prop-types';

import { wgs84ToRd } from '../../../shared/services/coordinate-reference-system';
import MapSearchResultsCategory from './map-search-results-category/MapSearchResultsCategory';
import Notification from '../../../shared/components/notification/Notification';
import MaximizeIcon from '../../../../public/images/icon-maximize.svg';

const MapSearchResults = ({
  resultLimit,
  location,
  missingLayers,
  onItemClick,
  panoUrl,
  results,
  onMaximize,
  onPanoPreviewClick
}) => {
  const rdCoordinates = wgs84ToRd(location);

  const limitResults = (categories) => categories.map((category) => ({
    ...category,
    results: category.results.slice(0, resultLimit),
    subCategories: limitResults(category.subCategories),
    showMore: category.results.length > resultLimit
  }));

  return (
    <section className="map-search-results">
      <header
        className={`
          map-search-results__header
          map-search-results__header--${panoUrl ? 'pano' : 'no-pano'}
        `}
      >
        <button
          className={`
            map-search-results__header-pano-button
            map-search-results__header-pano-button--${panoUrl ? 'enabled' : 'disabled'}
          `}
          disabled={!panoUrl}
          onClick={onPanoPreviewClick}
          title={panoUrl ? 'Bekijk Panorama view' : 'Geen Panoramabeeld beschikbaar'}
        >
          {panoUrl && (
            <img
              alt="Panoramabeeld"
              className="map-search-results__header-pano"
              height="292"
              src={panoUrl}
              width="438"
            />
          )}
          <div className="map-search-results__header-container">
            <h1 className="map-search-results__header-title">Resultaten</h1>
            <h2 className="map-search-results__header-subtitle">
              locatie {rdCoordinates.x.toFixed(2)}, {rdCoordinates.y.toFixed(2)}
            </h2>
          </div>
        </button>
      </header>
      <div className="map-search-results__scroll-wrapper">
        <ul className="map-search-results__list">
          {missingLayers && (
            <li>
              <Notification>Geen details beschikbaar van: {missingLayers}</Notification>
            </li>
          )}
          {
            limitResults(results).map((mainCategory) => (
              <MapSearchResultsCategory
                key={mainCategory.categoryLabel}
                category={mainCategory}
                onItemClick={onItemClick}
                onShowMoreClick={onMaximize}
              />
            ))
          }
        </ul>
        <footer className="map-search-results__footer">
          <button
            className="map-search-results__button"
            onClick={onMaximize}
            title="Volledig weergeven"
          >
            <MaximizeIcon className="map-search-results__button-icon" />
            Volledig weergeven
          </button>
        </footer>
      </div>
    </section>
  );
};

MapSearchResults.defaultProps = {
  resultLimit: 25
};

MapSearchResults.propTypes = {
  location: PropTypes.object, // eslint-disable-line
  missingLayers: PropTypes.string, // eslint-disable-line
  onItemClick: PropTypes.func.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired,
  resultLimit: PropTypes.number,
  panoUrl: PropTypes.string, // eslint-disable-line
  results: PropTypes.array // eslint-disable-line
};

export default MapSearchResults;
