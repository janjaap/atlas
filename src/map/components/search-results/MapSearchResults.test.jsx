import React from 'react';
import { shallow } from 'enzyme';

import MapSearchResults from './MapSearchResults';
import { wgs84ToRd } from '../../../shared/services/coordinate-reference-system';

jest.mock('../../../shared/services/coordinate-reference-system');

describe('MapSearchResults', () => {
  beforeEach(() => {
    wgs84ToRd.mockImplementation(() => ({ x: 1.12345, y: 654.321 }));
  });

  afterEach(() => {
    wgs84ToRd.mockReset();
  });

  it('should calculate RD coordinates', () => {
    const location = {
      latitude: 15.999,
      longitude: 329.123
    };
    const clickHandler = jest.fn();
    shallow(
      <MapSearchResults
        location={location}
        onItemClick={clickHandler}
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        results={[]}
      />
    );
    expect(wgs84ToRd).toHaveBeenCalledWith(location);
  });

  describe('rendering', () => {
    it('should render empty results', () => {
      const clickHandler = jest.fn();
      const wrapper = shallow(
        <MapSearchResults
          onItemClick={clickHandler}
          onMaximize={clickHandler}
          onPanoPreviewClick={clickHandler}
          results={[]}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should render results', () => {
      const count = 2;
      const location = {
        latitude: 15.999,
        longitude: 329.123
      };
      const clickHandler = jest.fn();
      const panoUrl = 'pano-url';
      const missingLayers = 'Layer 1, Layer 2';
      const results = [
        { label: 'label', categoryLabel: 'category', uri: 'result-uri-1', type: 'monumenten/monument', results: [], subCategories: [] },
        { label: 'label', categoryLabel: 'category', uri: 'result-uri-2', type: 'monumenten/monument', results: [], subCategories: [] }
      ];
      const wrapper = shallow(
        <MapSearchResults
          count={count}
          location={location}
          onItemClick={clickHandler}
          panoUrl={panoUrl}
          missingLayers={missingLayers}
          onMaximize={clickHandler}
          onPanoPreviewClick={clickHandler}
          results={results}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should filter external monumenten on pand results', () => {
      const count = 3;
      const location = {
        latitude: 15.999,
        longitude: 329.123
      };
      const clickHandler = jest.fn();
      const results = [
        { label: 'label', categoryLabel: 'category', uri: 'result-uri-1', type: 'bag/pand', results: [], subCategories: [] },
        { label: 'label', categoryLabel: 'category', uri: 'result-uri-2', type: 'foo', results: [], subCategories: [] },
        { label: 'label', categoryLabel: 'category', uri: 'result-uri-3', type: 'monumenten/monument', results: [], subCategories: [] }
      ];
      const wrapper = shallow(
        <MapSearchResults
          count={count}
          location={location}
          onItemClick={clickHandler}
          onMaximize={clickHandler}
          onPanoPreviewClick={clickHandler}
          results={results}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('should update results', () => {
      const count = 2;
      const location = {
        latitude: 15.999,
        longitude: 329.123
      };
      const clickHandler = jest.fn();
      const panoUrl = 'pano-url';
      const results = [
        { label: 'label', categoryLabel: 'category', uri: 'result-uri-1', type: 'bag/pand', results: [], subCategories: [] },
        { label: 'label', categoryLabel: 'category', uri: 'result-uri-2', type: 'bag/pand', results: [], subCategories: [] }
      ];
      const wrapper = shallow(
        <MapSearchResults
          count={count}
          location={location}
          onItemClick={clickHandler}
          panoUrl={panoUrl}
          onMaximize={clickHandler}
          onPanoPreviewClick={clickHandler}
          results={results}
        />
      );
      expect(wrapper).toMatchSnapshot();

      wrapper.setProps({
        count: 3,
        location: {
          latitude: 1,
          longitude: 0
        },
        panoUrl: 'other-pano-url',
        missingLayers: 'Layer 3',
        results: [
          { label: 'label', categoryLabel: 'category', uri: 'result-uri-1', type: 'bag/pand', results: [], subCategories: [] },
          { label: 'label', categoryLabel: 'category', uri: 'result-uri-2', type: 'bag/pand', results: [], subCategories: [] },
          { label: 'label', categoryLabel: 'category', uri: 'result-uri-3', type: 'bag/pand', results: [], subCategories: [] }
        ]
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
