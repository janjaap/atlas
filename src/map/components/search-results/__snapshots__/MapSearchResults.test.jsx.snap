// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapSearchResults rendering should filter external monumenten on pand results 1`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--no-pano
        "
  >
    <button
      className="
            map-search-results__header-pano-button
            map-search-results__header-pano-button--disabled
          "
      disabled={true}
      onClick={[MockFunction]}
      title="Geen Panoramabeeld beschikbaar"
    >
      <div
        className="map-search-results__header-container"
      >
        <h1
          className="map-search-results__header-title"
        >
          Resultaten
        </h1>
        <h2
          className="map-search-results__header-subtitle"
        >
          locatie 
          1.12
          , 
          654.32
        </h2>
      </div>
    </button>
  </header>
  <div
    className="map-search-results__scroll-wrapper"
  >
    <ul
      className="map-search-results__list"
    >
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "bag/pand",
            "uri": "result-uri-1",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "foo",
            "uri": "result-uri-2",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "monumenten/monument",
            "uri": "result-uri-3",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <Component
          className="map-search-results__button-icon"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapSearchResults rendering should render empty results 1`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--no-pano
        "
  >
    <button
      className="
            map-search-results__header-pano-button
            map-search-results__header-pano-button--disabled
          "
      disabled={true}
      onClick={[MockFunction]}
      title="Geen Panoramabeeld beschikbaar"
    >
      <div
        className="map-search-results__header-container"
      >
        <h1
          className="map-search-results__header-title"
        >
          Resultaten
        </h1>
        <h2
          className="map-search-results__header-subtitle"
        >
          locatie 
          1.12
          , 
          654.32
        </h2>
      </div>
    </button>
  </header>
  <div
    className="map-search-results__scroll-wrapper"
  >
    <ul
      className="map-search-results__list"
    />
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <Component
          className="map-search-results__button-icon"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapSearchResults rendering should render results 1`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--pano
        "
  >
    <button
      className="
            map-search-results__header-pano-button
            map-search-results__header-pano-button--enabled
          "
      disabled={false}
      onClick={[MockFunction]}
      title="Bekijk Panorama view"
    >
      <img
        alt="Panoramabeeld"
        className="map-search-results__header-pano"
        height="292"
        src="pano-url"
        width="438"
      />
      <div
        className="map-search-results__header-container"
      >
        <h1
          className="map-search-results__header-title"
        >
          Resultaten
        </h1>
        <h2
          className="map-search-results__header-subtitle"
        >
          locatie 
          1.12
          , 
          654.32
        </h2>
      </div>
    </button>
  </header>
  <div
    className="map-search-results__scroll-wrapper"
  >
    <ul
      className="map-search-results__list"
    >
      <li>
        <Notification
          className=""
          level="info"
        >
          Geen details beschikbaar van: 
          Layer 1, Layer 2
        </Notification>
      </li>
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "monumenten/monument",
            "uri": "result-uri-1",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "monumenten/monument",
            "uri": "result-uri-2",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <Component
          className="map-search-results__button-icon"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapSearchResults rendering should update results 1`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--pano
        "
  >
    <button
      className="
            map-search-results__header-pano-button
            map-search-results__header-pano-button--enabled
          "
      disabled={false}
      onClick={[MockFunction]}
      title="Bekijk Panorama view"
    >
      <img
        alt="Panoramabeeld"
        className="map-search-results__header-pano"
        height="292"
        src="pano-url"
        width="438"
      />
      <div
        className="map-search-results__header-container"
      >
        <h1
          className="map-search-results__header-title"
        >
          Resultaten
        </h1>
        <h2
          className="map-search-results__header-subtitle"
        >
          locatie 
          1.12
          , 
          654.32
        </h2>
      </div>
    </button>
  </header>
  <div
    className="map-search-results__scroll-wrapper"
  >
    <ul
      className="map-search-results__list"
    >
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "bag/pand",
            "uri": "result-uri-1",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "bag/pand",
            "uri": "result-uri-2",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <Component
          className="map-search-results__button-icon"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;

exports[`MapSearchResults rendering should update results 2`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--pano
        "
  >
    <button
      className="
            map-search-results__header-pano-button
            map-search-results__header-pano-button--enabled
          "
      disabled={false}
      onClick={[MockFunction]}
      title="Bekijk Panorama view"
    >
      <img
        alt="Panoramabeeld"
        className="map-search-results__header-pano"
        height="292"
        src="other-pano-url"
        width="438"
      />
      <div
        className="map-search-results__header-container"
      >
        <h1
          className="map-search-results__header-title"
        >
          Resultaten
        </h1>
        <h2
          className="map-search-results__header-subtitle"
        >
          locatie 
          1.12
          , 
          654.32
        </h2>
      </div>
    </button>
  </header>
  <div
    className="map-search-results__scroll-wrapper"
  >
    <ul
      className="map-search-results__list"
    >
      <li>
        <Notification
          className=""
          level="info"
        >
          Geen details beschikbaar van: 
          Layer 3
        </Notification>
      </li>
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "bag/pand",
            "uri": "result-uri-1",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "bag/pand",
            "uri": "result-uri-2",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
      <MapSearchResultsCategory
        category={
          Object {
            "categoryLabel": "category",
            "label": "label",
            "results": Array [],
            "showMore": false,
            "subCategories": Array [],
            "type": "bag/pand",
            "uri": "result-uri-3",
          }
        }
        isSubcategory={false}
        key="category"
        onItemClick={[MockFunction]}
        onShowMoreClick={[MockFunction]}
      />
    </ul>
    <footer
      className="map-search-results__footer"
    >
      <button
        className="map-search-results__button"
        onClick={[MockFunction]}
        title="Volledig weergeven"
      >
        <Component
          className="map-search-results__button-icon"
        />
        Volledig weergeven
      </button>
    </footer>
  </div>
</section>
`;
